libraries/AP_Baro/AP_Baro_FBM320.cpp

```diff
@@ -0,0 +1,222 @@
+
+	return true;
+}
+
+bool AP_Baro_FBM320::init()
+{
+    if (!dev | !dev->get_semaphore()->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+        return false;
+    }
+
+    dev->set_speed(AP_HAL::Device::SPEED_HIGH);
+
+    uint8_t whoami;
+    if (!dev->read_registers(FBM320_REG_ID, &whoami, 1) ||
+        whoami != FBM320_WHOAMI) {
+        // not a FBM320
+        dev->get_semaphore()->give();
+        return false;
+    }
+    printf("FBM320 ID 0x%x\n", whoami);
+
+    if (!read_calibration()) {
+        dev->get_semaphore()->give();
+        return false;        
+    }
+
+    dev->write_register(FBM320_REG_CMD, FBM320_CMD_READ_P);
+    
+    instance = _frontend.register_sensor();
+
+    dev->get_semaphore()->give();
+    
+    // request 50Hz update
+    dev->register_periodic_callback(20 * AP_USEC_PER_MSEC, FUNCTOR_BIND_MEMBER(&AP_Baro_FBM320::timer, void));
+
+    return true;
+}
+
+/*
+  calculate corrected pressure and temperature
+ */
+void AP_Baro_FBM320::calculate_PT(int32_t UT, int32_t UP, int32_t &pressure, int32_t &temperature)
+{
+    const struct fbm320_calibration &cal = calibration;
+	int32_t DT, DT2, X01, X02, X03, X11, X12, X13, X21, X22, X23, X24, X25, X26, X31, X32, CF, PP1, PP2, PP3, PP4;
+	
+	DT	=	((UT - 8388608) >> 4) + (cal.C0 << 4);
+	X01	=	(cal.C1 + 4459) * DT >> 1;
+	X02	=	((((cal.C2 - 256) * DT) >> 14) * DT) >> 4;
+	X03	=	(((((cal.C3 * DT) >> 18) * DT) >> 18) * DT);
+
+	temperature = ((2500 << 15) - X01 - X02 - X03) >> 15;
+
+	DT2	=	(X01 + X02 + X03) >> 12;	
+	X11	=	((cal.C5 - 4443) * DT2);
+	X12	=	(((cal.C6 * DT2) >> 16) * DT2) >> 2;
+	X13	=	((X11 + X12) >> 10) + ((cal.C4 + 120586) << 4);
+				
+	X21	=	((cal.C8 + 7180) * DT2) >> 10;
+	X22	=	(((cal.C9 * DT2) >> 17) * DT2) >> 12;
+	X23 =	(X22 >= X21) ? (X22 - X21) : (X21 - X22);
+
+	X24	=	(X23 >> 11) * (cal.C7 + 166426);
+	X25	=	((X23 & 0x7FF) * (cal.C7 + 166426)) >> 11;
+	X26 =	(X21 >= X22) ? (((0 - X24 - X25) >> 11) + cal.C7 + 166426) : (((X24 + X25) >> 11) + cal.C7 + 166426);
+
+	PP1	=	((UP - 8388608) - X13) >> 3;
+	PP2	=	(X26 >> 11) * PP1;
+	PP3	=	((X26 & 0x7FF) * PP1) >> 11;
+	PP4	=	(PP2 + PP3) >> 10;
+				
+	CF	=	(2097152 + cal.C12 * DT2) >> 3;
+	X31	=	(((CF * cal.C10) >> 17) * PP4) >> 2;
+	X32	=	(((((CF * cal.C11) >> 15) * PP4) >> 18) * PP4);
+
+	pressure = ((X31 + X32) >> 15) + PP4 + 99880;
+}
+
+//  acumulate a new sensor reading
+void AP_Baro_FBM320::timer(void)
+{
+    uint8_t buf[3];
+
+    if (!dev->read_registers(0xF6, buf, sizeof(buf))) {
+        return;
+    }
+	int32_t value = ((uint32_t)buf[0] << 16) | ((uint32_t)buf[1] << 8) | (uint32_t)buf[2];
+
+    if (step == 0) {
+        value_T = value;
+    } else {
+        int32_t pressure, temperature;
+        calculate_PT(value_T, value, pressure, temperature);
+        if (_sem->take_nonblocking()) {
+            pressure_sum += pressure;
+            temperature_sum += temperature*0.01;
+            count++;
+            _sem->give();
+        }
+    }
+
+    if (step++ == 5) {
```

Comment:
pet peeve here. If this is where step is incrementing, and it's being compared to in the same line and the inc is happening first, can we have this as ++step instead of step++ just so it's totally obvious?

Reply:
also, I have a phobia of unforeseen "magical bumps" in the increments so I always prefer >=5 instead of ==5 just as a layer of protection. It also tells us that it should never exceed 5.

Reply:
changed to >=
didn't change to ++step. I prefer it how it is.